packages/web/src/components/gcds-link/gcds-link.tsx

```diff
@@ -0,0 +1,220 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Watch,
+  Prop,
+  State,
+  h,
+} from '@stencil/core';
+import { assignLanguage, observerConfig } from '../../utils/utils';
+import { inheritAttributes } from '../../utils/utils';
+import i18n from './i18n/i18n';
+
+@Component({
+  tag: 'gcds-link',
+  styleUrl: 'gcds-link.css',
+  shadow: true,
+})
+export class GcdsLink {
+  @Element() el: HTMLElement;
+
+  private shadowElement?: HTMLElement;
+
+  /**
+   * Props
+   */
+
+  /**
+   * Set the link size
+   */
+  @Prop({ mutable: true }) size: 'regular' | 'small' = 'regular';
+
+  @Watch('size')
+  validateSize(newValue: string) {
+    const values = ['regular', 'small'];
+
+    if (!values.includes(newValue)) {
+      this.size = 'regular';
+    }
+  }
+
+  // TODO: Do we need to validate display, rel, target, etc?
+
+  private handleClick = (e: Event) => {
+    if (this.clickHandler) {
+      this.clickHandler(e);
```

Comment:
Forgot this one last thing, can we remove the clickHandler? It's one thing I want to go through all the components and redo so no need to have it here. 

When a click event happens we will need to emit the custom events like `this.gcdsClick.emit();`.

Reply:
I think so! @melaniebmn I know we talked about keeping the click handler, would you mind if we took it out?

Reply:
@melaniebmn also apologies if I'm misunderstanding what we talked about and just mistakenly kept the clickhandler

Reply:
All good, you can take it out @daine :)